src/components/cards/DiseaseCardCompact.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { FunctionComponent, FC, Fragment } from "react";
+import { Card } from "franklin-sites";
+import { Context, ContextObj } from "../../types/context";
+import { createTableLink } from "../utils";
+import { DiseaseData } from "./DiseaseCard";
+import { useHistory } from "react-router";
+
+import "./DiseaseCardCompact.css";
+
+export function getAllItems(
+  diseaseItem: DiseaseData,
+  keyName: keyof DiseaseData,
+  totalItems = new Set()
+): any {
+  const items = diseaseItem[keyName] as any[];
+  if (items) {
+    items.forEach((item: any) => totalItems.add(item));
+  }
+  if (diseaseItem.children) {
+    diseaseItem.children.forEach((childDisease) => {
+      getAllItems(childDisease, keyName, totalItems);
+    });
+  }
+  return Array.from(totalItems);
+}
+
+const generateDiseaseLinks = (diseaseItem: DiseaseData) => {
+  const diseaseLinks = [];
+  const { diseaseId } = diseaseItem;
+
+  const allProts = getAllItems(diseaseItem, "proteins");
+  const allDrugs = getAllItems(diseaseItem, "drugs");
+  const allVariants = getAllItems(diseaseItem, "variants");
+
+  if (allProts && allProts.length > 0) {
+    diseaseLinks.push(
+      createTableLink(diseaseId, Context.PROTEIN, allProts.length)
+    );
+  }
+  if (allDrugs && allDrugs.length > 0) {
+    diseaseLinks.push(
+      createTableLink(diseaseId, Context.DRUG, allDrugs.length)
+    );
+  }
+  if (allVariants && allVariants.length > 0) {
+    diseaseLinks.push(
+      createTableLink(diseaseId, Context.VARIANT, allVariants.length)
+    );
+  }
+  return diseaseLinks;
+};
+
+const generateSpacer = (size: number) => {
+  return "--".repeat(size);
```

Comment:
I would maybe use a single horizontal line character `─` as it looks better than two dashes (they have some space between them)
Or an other character from [](https://en.wikipedia.org/wiki/Box-drawing_character) like `└` for example